Sources/Utility/ArgumentParser.swift

```diff
@@ -320,6 +325,13 @@ public final class ArgumentParser: ArgumentParserProtocol {
         argumentsIterator = args.makeIterator()
 
         while let arg = argumentsIterator.next() {
+            // If argument is help then just print usage and exit.
+            if arg == "-h" || arg == "--help" {
+                printUsage(on: usageStream)
+                // Use throw to bail out when help is encountered so we don't have to
+                // worry about handling the help modes manually.
+                throw ArgumentParserError.helpEncountered
```

Comment:
@ddunbar what do you think about exploiting `throw` to print help automatically, Other option is clients handles it manually.